serenity/agent_utils.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef SERENITY_AGENT_UTILS_HPP
+#define SERENITY_AGENT_UTILS_HPP
+
+#include <mutex>  // NOLINT [build/c++11]
+#include <string>
+
+#include "curl_easy.h"  // NOLINT [build/include]
+
+#include "glog/logging.h"
+
+#include "rapidjson/document.h"
+
+#include "stout/error.hpp"
+#include "stout/nothing.hpp"
+#include "stout/option.hpp"
+#include "stout/try.hpp"
+
+namespace mesos {
+namespace serenity {
+
+class AgentInfo {
+ public:
+  static Try<std::string> GetHostName() {
+    std::lock_guard<std::mutex> lock(connectionMutex);
+    if (hostname.isNone()) {
+      Try<Nothing> result = FillAgentInfo();
+      if (result.isError()) {
+        return result.error();
+      }
+    }
+    return hostname.get();
+  }
+
+
+  static Try<std::string> GetAgentId() {
+    std::lock_guard<std::mutex> lock(connectionMutex);
+    if (agentId.isNone()) {
+      Try<Nothing> result = FillAgentInfo();
+      if (result.isError()) {
+        return result.error();
+      }
+    }
+    return agentId.get();
+  }
+
+ protected:
+  static Try<Nothing> FillAgentInfo() {
+    Try<std::string> result = GetStateFromAgent();
+    if (result.isError()) {
+      return Error(result.error());
+    } else {
+      rapidjson::Document doc;
+      doc.Parse(result.get().c_str());
+
+      hostname = (doc["hostname"]).GetString();
+      agentId = (doc["id"]).GetString();
+
+      return Nothing();
+    }
+  }
+
+
+  static Try<std::string> GetStateFromAgent(
+      std::string agentUrl = "http://localhost:5051/state.json") {
```

Comment:
We could parametrize port here. For demo it would be always 5051.. Maybe add some comment (TODO(..))?